### PR TITLE
Remove cache_data and random_password in favor of those in extlib

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -39,6 +39,10 @@
     {
       "name": "puppetlabs/stdlib",
       "version_requirement": ">= 4.2.0 < 5.0.0"
+    },
+    {
+      "name": "puppetlabs/extlib",
```
Comment (domcleal) [inline metadata.json]: puppet/extlib

Comment (ehelms) [inline metadata.json]: Updated

**Comment by domcleal** [comment]
Seems to work perfectly, thanks.  Would you mind making the same change to foreman_proxy, for its oauth* parameters?

**Comment by ehelms** [comment]
Here is the foreman_proxy update -- https://github.com/theforeman/puppet-foreman_proxy/pull/195

**Comment by stbenjam** [comment]
Will the major be bumped to 5 for this change?

**Comment by ekohl** [comment]
I think it should because the APIs are incompatible.

**Comment by domcleal** [comment]
Yep, true, that could be done as part of this PR.  :+1: otherwise.

**Comment by ekohl** [comment]
:+1:

**Comment by domcleal** [comment]
Thanks @ehelms, merged.  I bumped the version in a separate commit.

**Pull request merged**